### PR TITLE
Add an option to streamplot to manually specify the seed points.

### PR DESCRIPTION
**Pull request by Cadair**
The idea of this is to be able set the seed points for the streamlines manually. I needed this because I was using it for magnetic field visualisation, which is more physical if you have a few long lines rather than lots of short ones, and I also wanted to seed them at the point in the domain which lead to the best integration.

This is a first pass, it seems to work for me at the moment. I can tidy it up and add tests if people think it is worth while.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4342,7 +4342,7 @@ def stackplot(self, x, *args, **kwargs):
 
     def streamplot(self, x, y, u, v, density=1, linewidth=None, color=None,
                    cmap=None, norm=None, arrowsize=1, arrowstyle='-|>',
-                   minlength=0.1, transform=None, zorder=1):
+                   minlength=0.1, start_points=None, transform=None, zorder=1):
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: new kwargs need to go last for back compatibility.

File: lib/matplotlib/streamplot.py [hunk 2]
```diff
@@ -52,6 +52,8 @@ def streamplot(axes, x, y, u, v, density=1, linewidth=None, color=None,
         See :class:`~matplotlib.patches.FancyArrowPatch`.
     *minlength* : float
         Minimum length of streamline in axes coordinates.
+    *start_points*: Nx2 array
+        Coordinates of starting points for the streamlines. In array coordinates.
```
Comment (efiring) [inline lib/matplotlib/streamplot.py]: What do you mean by "array coordinates"?  I would expect starting points to be specified in data coordinates--that is, in the same coordinates as your x and y input arrays.

Comment (Cadair) [inline lib/matplotlib/streamplot.py]: oh yeah, that's an oversight, I have not modified that since I worked out the data-array coordinate conversion.

**Comment by efiring** [comment]
This was a good idea; thanks for doing it.

**Comment by Cadair** [comment]
not a problem, been meaning to do it for over a year!!

Thanks

**Pull request merged**